packages/ra-ui-materialui/src/list/DatagridRow.js

```diff
@@ -177,9 +177,10 @@ export class DatagridRow extends Component {
                         ) : null
                     )}
                 </TableRow>
-                {expand && expanded && (
+                {// TODO: add proper aria attributes (aria-expanded)
+                expand && expanded && (
                     <TableRow key={`${id}-expand`}>
-                        <TableCell colSpan={colSpan} role="expand-content">
```

Comment:
why ?

Reply:
Same reason. However there are other ways but it deserves a dedicated PR. I'll open an issue to track it

Reply:
Fixed it at it was needed for the tests